src/PoolManager.sol

```diff
@@ -188,7 +188,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
             })
         );
 
-        _accountPoolBalanceDelta(key, delta);
+        _accountPoolBalanceDelta(key, delta + feeDelta);
```

Comment:
another option is keeping the credit logic inside `Pool.sol` and this PR mostly becomes bubbling up `feeDelta` from Pool library --> PoolManager --> caller